.teamcity/Gradle_Promotion/buildTypes/PromoteRelease.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package Gradle_Promotion.buildTypes
+
+import jetbrains.buildServer.configs.kotlin.v2018_2.ParameterDisplay
+
+open class PromoteRelease(task: String, requiredConfirmationCode: String, branch: String = "release", init: PromoteRelease.() -> Unit = {}) : BasePromoteSnapshot(
+    branch = branch,
+    task = task,
+    triggerName = "ReadyforRelease",
+    gitUserEmail = "%gitUserEmail%",
+    gitUserName = "%gitUserName%",
+    extraParameters = "-PconfirmationCode=%confirmationCode%"
+) {
+    init {
+        params {
+            text("gitUserEmail", "", label = "Git user.email Configuration", description = "Enter the git 'user.email' configuration to commit change under", display = ParameterDisplay.PROMPT, allowEmpty = true)
+            text("confirmationCode", "", label = "Confirmation Code", description = "Enter the value '$requiredConfirmationCode' (no quotes) to confirm the promotion", display = ParameterDisplay.PROMPT, allowEmpty = false)
+            text("gitUserName", "", label = "Git user.name Configuration", description = "Enter the git 'user.name' configuration to commit change under", display = ParameterDisplay.PROMPT, allowEmpty = true)
+        }
+        this.init()
```

Comment:
Why do we need this `init` parameter in superclasses. Can't we just add an `init { }` to the subclasses if needed?

Reply:
I guess we can. Do you know why `BuildType` has this `init` constructor parameter?

Reply:
I removed most of the `init` parameters and renamed the classes.